src/components/ResultList.js

```diff
@@ -1,36 +1,32 @@
 import React from 'react'
 import { Text, View, StyleSheet, FlatList, TouchableOpacity } from 'react-native'
-import { withNavigation } from 'react-navigation'
 import ResultCard from './ResultCard'
 import ResultListSkel from './ResultListSkel'
 
 
-const ResultList = ({title, results, ready, navigation}) => {
-  return (
-    <View style={styles.container}>
-      <Text style={styles.title}>{title}</Text>
-      <View style={styles.box}>
-        {ready === false 
-          ? <ResultListSkel />
-          : <FlatList
-              horizontal
-              data={results}
-              keyExtractor={result => result.id}
-              showsHorizontalScrollIndicator={false}
-              renderItem={({item}) => {
-                return (
-                  <TouchableOpacity
-                    onPress={() => navigation.navigate('MoreInfo', { id: item.id })}
-                  >
-                    <ResultCard result={item} />
-                  </TouchableOpacity>
-                )
-              }}
-            />}
-      </View>
+const ResultList = ({title, results, ready, onPress}) => (
+  <View style={styles.container}>
+    <Text style={styles.title}>{title}</Text>
+    <View style={styles.box}>
+      {(!ready)  
+        ? <ResultListSkel />
+        : <FlatList
+            horizontal
+            data={results}
+            keyExtractor={result => result.id}
+            showsHorizontalScrollIndicator={false}
+            renderItem={({item}) => (
+              <TouchableOpacity 
+              onPress={() => onPress(item.id)}
+              >
+                <ResultCard result={item} />
+              </TouchableOpacity>
+            )}
```

Comment:
ResultsList - one more mistake, navigation to another screen was performed from nested component not from parent one. Change wrong logic of ResultsList, now it doesn't perform screen change directly, but only send callback with 'id' of component that was pressed, to the parent component, and parent component performs a screen change from the top level.